### PR TITLE
Add autoConfigure option so user can disable broker autoconfiguration

### LINKED ISSUE #88: Controller-only mode support
Hi,

We are interested in using the native image to test a cluster with some controller-only nodes, so `process.roles=controller`. We have:
1. node-0=`broker,controller` 
2. node-1:`controller`
3. node-2:`controller`

Here's our `server.properties` for one of the controller-only containers:

```properties
node.id=1
offsets.topic.num.partitions=1
metrics.jmx.exclude=.*
listeners=CONTROLLER\://0.0.0.0\:9091
transaction.state.log.replication.factor=1
group.initial.rebalance.delay.ms=0
offsets.topic.replication.factor=1
transaction.state.log.min.isr=1
controller.quorum.voters=0@//broker-0\:9091,1@//broker-1\:9091,2@//broker-2\:9091
controller.listener.names=CONTROLLER
process.roles=controller
broker.id=1
early.start.listeners=CONTROLLER
listener.security.protocol.map=CONTROLLER\:PLAINTEXT
```

When we start it up we see a failure like:

``` shell
docker run -it --rm -p 19092:9091 \
   -v $(pwd):/conf \
   -e SERVER_PROPERTIES_FILE=/conf/server.properties \
   quay.io/ogunalp/kafka-native:latest
```
```
2023-08-24 02:21:49,407 ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.IllegalArgumentException: No security protocol defined for listener PLAINTEXT
	at kafka.cluster.EndPoint$.$anonfun$createEndPoint$2(EndPoint.scala:49)
	at scala.collection.immutable.Map$Map1.getOrElse(Map.scala:248)
	at kafka.cluster.EndPoint$.securityProtocol$1(EndPoint.scala:49)
	at kafka.cluster.EndPoint$.createEndPoint(EndPoint.scala:54)
	at kafka.utils.CoreUtils$.$anonfun$listenerListToEndPoints$6(CoreUtils.scala:270)
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:100)
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87)
	at scala.collection.mutable.ArraySeq.map(ArraySeq.scala:37)
	at kafka.utils.CoreUtils$.listenerListToEndPoints(CoreUtils.scala:270)
	at kafka.server.KafkaConfig.effectiveAdvertisedListeners(KafkaConfig.scala:2034)
	at kafka.server.KafkaConfig.validateValues(KafkaConfig.scala:2148)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:2107)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1534)
	at kafka.server.KafkaConfig.fromProps(KafkaConfig.scala:1457)
	at com.ozangunalp.kafka.server.EmbeddedKafkaBroker.start(EmbeddedKafkaBroker.java:182)
	at com.ozangunalp.kafka.server.Startup.startup(Startup.java:36)
	at com.ozangunalp.kafka.server.Startup_Observer_startup_ba142133319fbb85a5c3c9a901d8261b72ab2d8b.notify(Unknown Source)
```

It looks like advertised.listeners is being defaulted [here](https://github.com/ozangunalp/kafka-native/blob/5e82657def7a8b594e270414ec9c4a167e8cf61c/kafka-server/src/main/java/com/ozangunalp/kafka/server/BrokerConfig.java#L64C19-L64C19) if we don't specify it, using PLAINTEXT as listener name by default. So it fails when it tries to lookup the security protocol.

I tried setting advertised listeners to match `listeners` via the env var, and then get a failure:

``` shell
docker run -it --rm -p 19092:9091 \
    -v $(pwd):/conf  \
    -e KAFKA_ADVERTISED_LISTENERS="CONTROLLER://0.0.0.0:9091" \
    -e SERVER_PROPERTIES_FILE=/conf/server.properties \
    quay.io/ogunalp/kafka-native:latest
```

```
2023-08-24 02:23:05,459 ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.IllegalArgumentException: requirement failed: The advertised.listeners config must be empty when process.roles=controller
	at scala.Predef$.require(Predef.scala:337)
	at kafka.server.KafkaConfig.validateValues(KafkaConfig.scala:2222)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:2107)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1534)
	at kafka.server.KafkaConfig.fromProps(KafkaConfig.scala:1457)
	at com.ozangunalp.kafka.server.EmbeddedKafkaBroker.start(EmbeddedKafkaBroker.java:182)
	at com.ozangunalp.kafka.server.Startup.startup(Startup.java:36)
	at com.ozangunalp.kafka.server.Startup_Observer_startup_ba142133319fbb85a5c3c9a901d8261b72ab2d8b.notify(Unknown Source)
```

And I couldn't see a way to unset it, smallrye complains if you set the env var to an empty string.
``` shell
docker run -it --rm -p 19092:9091 \
    -v $(pwd):/conf  \
    -e KAFKA_ADVERTISED_LISTENERS="" \
    -e SERVER_PROPERTIES_FILE=/conf/server.properties \
    quay.io/ogunalp/kafka-native:latest
```

```
2023-08-24 02:10:12,680 ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.util.NoSuchElementException: SRCFG00040: The config property kafka.advertised.listeners is defined as the empty String ("") which the following Converter considered to be null: io.smallrye.config.Converters$BuiltInConverter
	at io.smallrye.config.SmallRyeConfig.convertValue(SmallRyeConfig.java:300)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:242)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:168)
	at com.ozangunalp.kafka.server.BrokerConfig.providedConfig(BrokerConfig.java:190)
```

So it appears advertised.listeners is incompatible with controller only mode, would it make sense to add a check [here](https://github.com/ozangunalp/kafka-native/blob/5e82657def7a8b594e270414ec9c4a167e8cf61c/kafka-server/src/main/java/com/ozangunalp/kafka/server/BrokerConfig.java#L63) to make it not set up the default `advertised.listeners` if the only role is `controller`?

### PR DESCRIPTION
**Pull request by robobario**
To disable auto configuration of the broker, set system property `-Dserver.auto-configure=false` or environment variable `SERVER_AUTO_CONFIGURE=false`. This should be used in combination with `SERVER_PROPERTIES_FILE`.

Why:
Users may wish to supply all the configuration themselves to test odd scenarios like a node in controller-only mode, where advertised listeners should not be set in the server properties. Configurations like this may be outside of the scope of what the auto configuration is trying to cover. In this case we can provide an escape hatch for users to supply all the configuration.

Resolves #88

### CODE REVIEW

**Comment by robobario** [comment]
Hi @ozangunalp just bumping this, keen to get a review, thank you

File: kafka-native-test-container/src/test/java/com/ozangunalp/kafka/test/container/KafkaNativeContainerIT.java [hunk 1]
```diff
@@ -271,4 +310,14 @@ void testKraftClusterOneController() throws Exception {
         }
     }
 
+    MountableFile classpathResourceWithStringsReplaced(String resourceName, Path tempDir, Map<String, String> replacements) throws IOException {
+        MountableFile serverPropertiesFile = MountableFile.forClasspathResource(resourceName);
+        String props = Files.readString(Path.of(serverPropertiesFile.getFilesystemPath()), StandardCharsets.UTF_8);
+        for (Map.Entry<String, String> entry : replacements.entrySet()) {
+            props = props.replaceAll(quote(entry.getKey()), entry.getValue());
+        }
+        Path out = tempDir.resolve("controller_only.properties");
```
Comment (k-wall) [inline kafka-native-test-container/src/test/java/com/ozangunalp/kafka/test/container/KafkaNativeContainerIT.java:319]: "controller_only.properties" -> resourceName?

File: README.md [hunk 2]
```diff
@@ -100,6 +100,7 @@ Following configuration options are available:
 | `server.host`                 | Hostname of listeners                                     | `` (empty string)  |
 | `server.cluster-id`           | Provide `cluster-id`, generated if empty                  |                    |
 | `server.properties-file`      | Path to `server.properties` file                          |                    |
+| `server.auto-configure `      | Automatically configure server properties                 | true               |
```
Comment (k-wall) [inline README.md:103]: document that if set false, only server.properties-file is respected.

**Comment by k-wall** [comment]
feature looks good to me, just a few nits.

File: kafka-native-test-container/src/test/java/com/ozangunalp/kafka/test/container/KafkaNativeContainerIT.java [hunk 3]
```diff
@@ -271,4 +310,14 @@ void testKraftClusterOneController() throws Exception {
         }
     }
 
+    MountableFile classpathResourceWithStringsReplaced(String resourceName, Path tempDir, Map<String, String> replacements) throws IOException {
```
Comment (ozangunalp) [inline kafka-native-test-container/src/test/java/com/ozangunalp/kafka/test/container/KafkaNativeContainerIT.java:313]: I'd change the `withServerPropertes` method to take a `Transferable` instead. That way you don't need to do this trick, You can just build a `Transferable.of` with the string and pass that.

File: kafka-server/src/main/java/com/ozangunalp/kafka/server/EmbeddedKafkaBroker.java [hunk 4]
```diff
@@ -172,9 +186,13 @@ public synchronized EmbeddedKafkaBroker start() {
...
 
-        BrokerConfig.providedConfig(brokerConfig);
-        BrokerConfig.defaultStaticConfig(brokerConfig);
-        BrokerConfig.defaultCoreConfig(brokerConfig, host, kafkaPort, internalPort, controllerPort, defaultProtocol);
+        if (autoConfigure) {
+            BrokerConfig.providedConfig(brokerConfig);
+            BrokerConfig.defaultStaticConfig(brokerConfig);
+            BrokerConfig.defaultCoreConfig(brokerConfig, host, kafkaPort, internalPort, controllerPort, defaultProtocol);
+        } else {
+            LOGGER.info("skipping auto-configuration of server");
```
Comment (ozangunalp) [inline kafka-server/src/main/java/com/ozangunalp/kafka/server/EmbeddedKafkaBroker.java:194]: It may be better to log the auto-configuration instead.

**Review by ozangunalp — CHANGES_REQUESTED** [review]

File: kafka-native-test-container/src/test/java/com/ozangunalp/kafka/test/container/KafkaNativeContainerIT.java (hunk 1 above)
Comment (robobario) [reply kafka-native-test-container/src/test/java/com/ozangunalp/kafka/test/container/KafkaNativeContainerIT.java]: the method is removed as part of refactoring withServerProperties to take Transferable. The test can operate on String now without messing with temporary files.

**Review by ozangunalp — APPROVED** [review]
Looks great! Thanks

**Pull request merged**